Scottish-Tech-Army/Volunteer-app
# Feature/SVA-426 Create login screen

- [Description](#description)
  - [Type of change](#type-of-change)
- [Testing locally](#testing-locally)
- [Checklist](#checklist)

# Description

- Creates a login screen with redirect action
- It is behind a feature flag so it is only visible if `login` is enabled in your local environment variables. 
- The login screen can be accessed from any `Project Details` screen by pressing the `Register Interest` button. 
- It will test if the user is logged in and open the new `Login` screen. Upon entering email (can be any string at the moment) it will redirect to the `Register Interest` screen if the previous page was a `Project Details` screen, if not it will go back to previous screen (ie settings if the login button was used) 

**NB** the authenticated state is not persisted, any time a button or link is used that tests if the user is authenticated the result will be that the user is not. This is because the login is not verified with an authentication code, which is blocked until the backend development and authentication journey development is completed. 

Also, small detail - the 'STA pencils' image in the top right is not behind the phone's status bar because we need a full screen background image for that. We can ask design for one when we update the login screen once the full login authentication journey can be implemented.  

Fixes # [SVA-426](https://sta2020.atlassian.net/browse/SVA-426)

## Type of change

Please delete options that are not relevant

- [] Bug fix (non-breaking change which fixes an issue)
- [] Breaking change(fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

# Testing locally

- add `login` to FEATURES_ENABLED in your local environment variables `.env` file so you can see the new feature (no need for spaces between the list if you have more than one enabled). 
- start the app and navigate to a project details page and tap the `Register Interest` button
  - expect to be directed to the login screen as the app detects you're not an authenticated user.
-  enter your email address (any string will be accepted)
   - expect login action to be completed and to be redirected to the `Register Interest` screen. 

NB You are not actually logged in, so any time you press that Register Interest button it tests if you are authenticated and will return false, directing you to the login screen again. 

# Checklist

- [] I have made corresponding changes to the documentation (if required)
- [] I have addressed accessibility, if needed
- [] I have added tests that prove my fix is effective or that my feature works
- [] I have updated the version numbers in `package.json` files in the [app](DEPLOYMENT.md#app-deployment) and/or [api](DEPLOYMENT.md#api-deployment-on-aws) directories as needed
